### PR TITLE
Extracts UTF-8 valid length position to Bitstring class

### LINKED ISSUE #703: Refactor :unicode module - extract UTF-8 prefix validation to BitString
Extract duplicated UTF-8 validation logic into a centralized method in the Bitstring class, eliminating code duplication across multiple Unicode normalization functions.

  - Validate UTF-8 byte sequences
  - Finds the longest valid UTF-8 prefix
  - Detect truncated sequences (incomplete but potentially valid UTF-8)
  - Return `{validLength: number, isTruncated: boolean}`
  - Handles edge cases: overlong encodings, surrogates, invalid continuation bytes

### PR DESCRIPTION
**Pull request by mward-sudo**
Closes #703 

### Dependencies

- [ ] #710
- [ ] #715
- [ ] #716 
- [ ] #717 
- [ ] #718

_Please note that this PR includes commits from the PR(s) it is dependent upon. Once the dependent PR(s) are merged to the dev branch, then this PR will be rebased and will then only contain its own commits. This PR will remain in draft until that point._

## Summary by CodeRabbit

* **New Features**
  * Added UTF-8 decoding and validation utilities to improve handling of multi-byte and truncated sequences.

* **Refactor**
  * Centralized UTF-8 validation logic across Unicode operations for consistency and reduced duplication; normalization flows updated for NFC/NFD/NFKC/NFKD.

* **Tests**
  * Added extensive UTF-8 tests covering decoding, validation, truncation, edge cases, and boundary conditions.

### CODE REVIEW

File: assets/js/bitstring.mjs [hunk 1]
```diff
@@ -572,6 +614,67 @@ export default class Bitstring {
...
+    const availableBytes = bytes.length - start;
+    if (availableBytes >= expectedLength) return false;
+
+    // Check all available continuation bytes
+    for (let i = 1; i < availableBytes; i++) {
+      if (!$.isValidUtf8ContinuationByte(bytes[start + i])) return false;
+    }
+
+    return true;
+  }
```
[the remark this answers is not in the document]
Comment (mward-sudo) [reply assets/js/bitstring.mjs:678]: Internal callers only, validation not planned.

**Comment by mward-sudo** [comment]
@coderabbitai Don't do any further reviews on this PR while it is in draft.
